Assets/Scripts/ECS/Hybrid/Interaction.cs

```diff
@@ -17,8 +21,14 @@ public void Convert(Entity entity, EntityManager dstManager, GameObjectConversio
         /// Kills an animal.
         /// </summary>
         /// <param name="animal">The animal entity to kill</param>
-        public void Kill(Entity animal)
+        public void Kill(Entity animal, GameObject prefab)
```

Comment:
This goes for the "Eat" method as well:
- The GameObject here is not a prefab
- The callers of this method will not have a reference to the GameObject ready
- The logic for destroying/disabling/recycling GameObjects should not be handled here. It would make more sense to do that in the same system that destroys the entities for example.